staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -148,6 +165,37 @@ func NewDefaultClientConfigLoadingRules() *ClientConfigLoadingRules {
 	}
 }
 
+func loadFileOrDir(fileOrDir string) ([]string, error) {
+	stat, err := os.Stat(fileOrDir)
+	if err != nil {
+		return []string{}, err
+	}
+	if stat.IsDir() {
+		return loadConfigDir(fileOrDir)
+	}
+	return []string{fileOrDir}, nil
+}
+
+func loadConfigDir(dir string) ([]string, error) {
+	if len(dir) == 0 {
+		return []string{}, nil
+	}
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return []string{}, err
+	}
+	result := []string{}
+	for _, file := range files {
+		// Skip hidden files
+		if strings.HasPrefix(file.Name(), ".") {
+			continue
+		}
+		result = append(result, path.Join(dir, file.Name()))
+	}
+	sort.Strings(result)
```

Comment:
@deads2k I would have to imagine that the call to `io.ReadDir(...)` uses the "os native' listing for whatever that's worth, so I'd suspect this sort is unnecessary (but it was requested).

I can add further documentation to the `kubectl` help docs if you would prefer.